### ISSUE TITLE
Customer related values are NULL for guests converted to customers after checkout.

### ISSUE THREAD

**Issue by jakwinkler** [magento/magento2#19166]
### Preconditions (*)

1. Magento 2.2.6 CE (OpenSource)
2. No checkout customizations, 

### Steps to reproduce (*)

1. Allow customers to create account after checkout 
2. Create few orders as guest customers and create an account after checkout process
3. Create a module with a plugin aroundSave for SalesOrder class
4. In aroundSave method, try to fetch customer name from order object
```
public function aroundSave(
        OrderResource $subject,
        \Closure $proceed,
        SalesOrder $order
    ) {
        [...]
        if ($order->getCustomerId()) {
            $customerName = $order->getCustomerName(); // returns NULL
        [...]
        }
    }
```

### Expected result (*)

1. customer name and all customer related values should not be NULL 

### Actual result (*)

1. customer name, customer last name are NULL 
![customer_data](https://user-images.githubusercontent.com/4256191/48338631-717efb00-e666-11e8-8904-418f074a1a54.png)
2. when you look at data by PMA, you can see when guest customer was converted to a registered customer, all customer related values in sales_order table are NULL

**Comment by magento-engcom-team** [comment]
Hi @qsolutions-pl. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@qsolutions-pl do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96273`, `MAGETWO-96274` were created

**Comment by magento-engcom-team** [comment]
Hi @Nazar65. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @qsolutions-pl. Thank you for your report. 
The issue has been fixed in magento/magento2#19191 by @Nazar65 in 2.3-develop branch
Related commit(s):
 - [ccba42089f1ca465cb43fc60339671db5e9a504a](https://github.com/magento/magento2/commit/ccba42089f1ca465cb43fc60339671db5e9a504a)
 - [f8a6d059fbe002e6bcc1a81ea9fdc73f069f97e2](https://github.com/magento/magento2/commit/f8a6d059fbe002e6bcc1a81ea9fdc73f069f97e2)
 - [7607c829e066b541bed53fea65d171c006834849](https://github.com/magento/magento2/commit/7607c829e066b541bed53fea65d171c006834849)
 - [560e1d734484b328dd5f5488e71493707e16d205](https://github.com/magento/magento2/commit/560e1d734484b328dd5f5488e71493707e16d205)
 - [125de74c601d736e0905e89cc74b4cdb1adc37cd](https://github.com/magento/magento2/commit/125de74c601d736e0905e89cc74b4cdb1adc37cd)
 - [5fb9c5265ac571cfec0a694d1b2d4308a57253f6](https://github.com/magento/magento2/commit/5fb9c5265ac571cfec0a694d1b2d4308a57253f6)
 - [e20c7efd4b8d61dac21a3d1691d6b2c950826d59](https://github.com/magento/magento2/commit/e20c7efd4b8d61dac21a3d1691d6b2c950826d59)
 - [7d22ae870c3b4e8224f620f877686678c0f2fa16](https://github.com/magento/magento2/commit/7d22ae870c3b4e8224f620f877686678c0f2fa16)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @qsolutions-pl. Thank you for your report. 
The issue has been fixed in magento/magento2#21325 by @Nazar65 in 2.2-develop branch
Related commit(s):
 - [9746716226a11b4f19a6e1351c4428df2dcc3742](https://github.com/magento/magento2/commit/9746716226a11b4f19a6e1351c4428df2dcc3742)
 - [abcf202ef6bc9c3d9b3276215126e7404b5326e7](https://github.com/magento/magento2/commit/abcf202ef6bc9c3d9b3276215126e7404b5326e7)
 - [80e5e39b25c0c5050c291f2c11d2f5cd2ffe7a7c](https://github.com/magento/magento2/commit/80e5e39b25c0c5050c291f2c11d2f5cd2ffe7a7c)
 - [41fc945ea5fd7eae7bf10f9aaa6954b18cba6e50](https://github.com/magento/magento2/commit/41fc945ea5fd7eae7bf10f9aaa6954b18cba6e50)
 - [79429edd3f660bac9c26e09afab5abefe6c8a178](https://github.com/magento/magento2/commit/79429edd3f660bac9c26e09afab5abefe6c8a178)

The fix will be available with the upcoming 2.2.9 release.
